pkg/controllers/provisioning/provisioner.go

```diff
@@ -227,13 +227,21 @@ func (p *Provisioner) NewScheduler(ctx context.Context, pods []*v1.Pod, stateNod
 			requirements.Add(instanceType.Requirements.Values()...)
 
 			for key, requirement := range requirements {
-				domains[key] = domains[key].Union(sets.NewString(requirement.Values()...))
+				if domains[key] == nil {
```

Comment:
I can see someone coming along later and asking why this was done. Do you mind commenting (linking the issue?) as well as giving a brief explanation of the performance improvement here?

Reply:
Added the requested comments, let me know if this is clear